sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -152,24 +158,23 @@ internal class AmqpConnectionScope : TransportConnectionScope
                 connectionFactory,
                 CloseConnection);
             TransactionController = new FaultTolerantAmqpObject<Controller>(
-                CreateControllerAsync,
-                CloseController);
+                async (timeout) =>
+                {
+                    var stopWatch = ValueStopwatch.StartNew();
+                    AmqpConnection connection = await ActiveConnection.GetOrCreateAsync(timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                    AmqpSession amqpSession = await CreateAndOpenSessionAsync(connection, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                    return await CreateControllerAsync(amqpSession, timeout.CalculateRemaining(stopWatch.GetElapsedTime())).ConfigureAwait(false);
+                },
+                controller => controller.Close());
```

Comment:
Should this be `SafeClose`?    Most of the other fault tolerant wrappers use the form that guarantees not throwing in their `Close` delegate.